File: SourceCode/Debug Helpers/open.fda.gov/ApiDalc/Enumerations/OpenFDAApiEndPoints.vb
Imports System.ComponentModel

Namespace Enumerations

    ''' <summary>
    ''' open.fda.gov API Endpoints
    ''' </summary>
    ''' <remarks></remarks>
    Public Enum OpenFDAApiEndPoints

        '<Description("drug/event")>
        '<DisplayNameAttribute("drug/event")>
        <Description("Drug Event")>
        <DefaultValueAttribute("drug/event")>
        DrugEvent

        <Description("Drug Label")>
        <DefaultValueAttribute("drug/label")>
        DrugLabel

        <Description("Drug Recall")>
        <DefaultValueAttribute("drug/enforcement")>
        DrugRecall

        <Description("Device Event")>
        <DefaultValueAttribute("device/event")>
        DeviceEvent

        <Description("Device Recall")>
        <DefaultValueAttribute("device/enforcement")>
        DeviceRecall

        <Description("Food Recall")>
        <DefaultValueAttribute("food/enforcement")>
        FoodRecall

    End Enum

End Namespace

File: SourceCode/AspCoreVersion/src/ShopAware.Core/Enumerations/OpenFdaApiEndPoints.cs
using System.ComponentModel;
using ShopAware.Core.Attributes;

namespace ShopAware.Core
{
    namespace Enumerations
    {
        /// [summary]
        /// open.fda.gov API Endpoints
        /// [/summary]
        /// [remarks][/remarks]
        public enum OpenFdaApiEndPoints
        {
            [Description("Drug Event")] [DefaultValue("drug/event")] DrugEvent,

            [Description("Drug Label")] [DefaultValue("drug/label")] DrugLabel,

            [Description("Drug Recall")] [DefaultValue("drug/enforcement")] DrugRecall,

            [Description("Device Event")] [DefaultValue("device/event")] DeviceEvent,

            [Description("Device Recall")] [DefaultValue("device/enforcement")] DeviceRecall,

            [Description("Food Recall")] [DefaultValue("food/enforcement")] FoodRecall
        }
    }
}